Mathlib/Topology/Category/FinTopCat.lean

```diff
@@ -0,0 +1,77 @@
+/-
+Copyright (c) 2024 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+import Mathlib.CategoryTheory.FintypeCat
+import Mathlib.Topology.Category.TopCat.Basic
+
+/-!
+# Category of finite topological spaces
+
+Definition of the category of finite topological spaces with the canonical
+forgetful functors.
+
+-/
+
+
+universe u
+
+open CategoryTheory
+
+/-- A bundled finite topological space. -/
+structure FinTopCat where
+  /-- carrier of a finite topological space. -/
+  α : Type u
+  [fintype : Fintype α]
```

Comment:
Is there a reason not to use `Finite α` instead of `Fintype`?

Reply:
Yes, because `FintypeCat` uses `Fintype`, so to have a computable forgetful functor to `FintypeCat`, this is needed.

Reply:
You can still define that functor -- you can always get a `Fintype` instance from a `Finite` instance (`Fintype.ofFinite`) -- it's just that `Finite` is a `Prop` and `Fintype` contains data. You can read the docstring [here](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Finite/Defs.html#Finite) for an explanation on when to use `Finite` and when to use `Fintype`

Reply:
I agree, but that's what would make the forgetful functor `noncomputable`. And, as I said, for consistency I think it makes sense to have the same typeclass assumption as in `FintypeCat`.

If I am understanding correctly, if I start with an object `X` of `FintypeCat` with a topological space instance and write `(forget₂ FinTopCat FintypeCat).obj (FinTopCat.of X)`, with the `Finite` approach I would get two (possibly non def-eq) `Fintype` instances on `X` (the one I started with and the one from `Fintype.ofFinite`).

Reply:
Note that I don't care about the `Fintype` instance, I just think that `FintypeCat` and `FinTopCat` should use the same notion of finiteness.

And I agree with you, I am in favor of using `Finite` instead of `Fintype`.

Reply:
I agree that `FinTopCat` and `FintypeCat` should have the same finiteness. Someone might want to open a Zulip discussion on whether `FintypeCat` should be changed to `FiniteCat` in case `FintypeCat` has some combinatorial applications that needs the `Fintype` instance.

Reply:
I agree that unless the name/definition of `FintypeCat` is changed, it is very much ok to keep the `Fintype` assumption here.